### PR TITLE
Add ability for worker to bind to local docker socket for testing exports

### PR DESCRIPTION
**Pull request by elliott-davis**
tenor-233124008

Closes: #117

Signed-off-by: Travis Elliott Davis <edavis@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-worker/src/runner/docker.rs [hunk 1]
```diff
@@ -145,7 +145,12 @@ impl<'a> DockerExporter<'a> {
         );
         cmd.env(DOCKER_HOST_ENVVAR, sock); // Use the job-specific `dockerd`
         cmd.stdout(Stdio::piped());
-        cmd.stderr(Stdio::piped());
+        match env::var_os(DEV_MODE) {
+            Some(_) => debug!("Not exporting in dev mode"),
```
Comment (chefsalim) [inline components/builder-worker/src/runner/docker.rs]: Sneaky :)  Maybe put in a comment that this part is a workaround for now until we solve the streaming log problem correctly?

File: components/builder-worker/src/runner/studio.rs [hunk 2]
```diff
@@ -104,7 +104,12 @@ impl<'a> Studio<'a> {
             }
         }
         cmd.stdout(Stdio::piped());
-        cmd.stderr(Stdio::piped());
+        match env::var_os(DEV_MODE) {
+            Some(_) => debug!("Studio not displaying stderr"),
```
Comment (chefsalim) [inline components/builder-worker/src/runner/studio.rs]: Same as above

File: components/builder-worker/src/runner/studio.rs [hunk 3]
```diff
@@ -97,43 +97,46 @@ impl<'a> Studio<'a> {
         cmd.env("PATH", env::var("PATH").unwrap_or(String::from(""))); // Sets `$PATH`
         cmd.env(NONINTERACTIVE_ENVVAR, "true"); // Disables progress bars
         cmd.env("TERM", "xterm-256color"); // Emits ANSI color codes
-        // propagate debugging environment variables into Airlock and Studio
+                                           // propagate debugging environment variables into Airlock and Studio
```
Comment (chefsalim) [inline components/builder-worker/src/runner/studio.rs]: Is the spacing intentional?

Comment (elliott-davis) [reply components/builder-worker/src/runner/studio.rs]: rustfmt did that

**Pull request merged**